plugins/inputs/mysql/mysql.go

```diff
@@ -588,9 +592,11 @@ func (m *Mysql) gatherSlaveStatuses(db *sql.DB, serv string, acc telegraf.Accumu
 	tags := map[string]string{"server": servtag}
 	fields := make(map[string]interface{})
 
-	// to save the column names as a field key
-	// scanning keys and values separately
-	if rows.Next() {
+	// for each channel record
+	for rows.Next() != false {
```

Comment:
The `!= false` is redundant, just use `for rows.Next() {`

Reply:
Removed the redundancy